engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentCreateProcessor.java

```diff
@@ -96,6 +96,9 @@ public void processRecord(final TypedRecord<DeploymentRecord> command) {
     try {
       createTimerIfTimerStartEvent(command);
     } catch (final RuntimeException e) {
+      // Make sure the cache does not contain any leftovers from this run (by hard resetting)
+      processState.clearCache();
+
       final String reason = String.format(COULD_NOT_CREATE_TIMER_MESSAGE, e.getMessage());
```

Comment:
:thought_balloon: Wondering we need this block at all, or whether we can handle this by our onProcessingError handle mechanism.

Nothing what you need to do now ofc